includes/collection/class-inbox.php

```diff
@@ -382,26 +382,31 @@ public static function get_by_guid_and_recipient( $guid, $user_id ) {
 	 * @return \WP_Post|false The primary inbox post, or false if no posts found.
 	 */
 	public static function deduplicate( $guid ) {
-		$duplicates = \get_posts(
-			array(
-				'post_type'      => self::POST_TYPE,
-				'guid'           => $guid,
-				'posts_per_page' => -1,
-				'post_status'    => 'any',
+		global $wpdb;
+
+		// Query for all posts with this GUID directly (get_posts doesn't supports guid parameter).
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		$post_ids = $wpdb->get_col(
```

Comment:
```suggestion
		$post_ids = $wpdb->get_col( // phpcs:ignore WordPress.DB.DirectDatabaseQuery
```
Does this work? Helps simplify the ignore and avoid the double inline comment.